packages/cli/src/lib/templates/currentUser.d.ts.template

```diff
@@ -0,0 +1,17 @@
+import '@redwoodjs/api'
+import '@redwoodjs/auth'
+
+import { getCurrentUser } from '../../api/src/lib/auth'
+
+type ThenArg<T> = T extends PromiseLike<infer U> ? U : T
+export type InferredCurrentUser = ThenArg<ReturnType<typeof getCurrentUser>>
+
+declare module '@redwoodjs/api' {
+  interface GlobalContext {
+    currentUser?: InferredCurrentUser
```

Comment:
Should this be optional?

Reply:
Yes I think so, because it might be an unauthenticated request. i.e. public graphql call

Reply:
I think it would be null in that case - but I may be wrong. I guess part of the problem is that we do not type guard what a user _should_ return in getCurrentUser.

Reply:
I think it's literally a case of currentUser not being defined if auth headers aren't present: https://github.com/redwoodjs/redwood/blob/17e0911ff5444ee7f1ffadd22359a83709511b13/packages/api/src/functions/graphql.ts#L48